tests/test_choices_new_api.py

```diff
@@ -0,0 +1,398 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from unittest import TestCase
+from unittest.mock import Mock
+
+from travertino.colors import NAMED_COLOR, rgb
+from travertino.constants import GOLDENROD, NONE, REBECCAPURPLE, TOP
+from travertino.declaration import BaseStyle, Choices, validated_property
+
+_DATACLASS_ARGS = (
+    {"kw_only": True} if "kw_only" in dataclass.__kwdefaults__ else {"init": False}
+)
```

Comment:
Is there any reason to do this, rather than `if sys.version_info < (3, 9)` check? The benefit of doing it with sys.version is that it will be automatically cleaned up by `pyupgrade` when 3.9 is deprecated.

Reply:
No particular reason, no — and I was in fact unaware of pyupgrade. Looks like it doesn't support ternary expressions, so I've made it a regular if/else.